middleware/logger/logger.go

```diff
@@ -195,84 +229,9 @@ func New(config ...Config) fiber.Handler {
 
 		// Loop over template tags to replace it with the correct value
 		_, err = tmpl.ExecuteFunc(buf, func(w io.Writer, tag string) (int, error) {
-			switch tag {
-			case TagTime:
-				return buf.WriteString(timestamp.Load().(string))
-			case TagReferer:
-				return buf.WriteString(c.Get(fiber.HeaderReferer))
-			case TagProtocol:
-				return buf.WriteString(c.Protocol())
-			case TagPid:
-				return buf.WriteString(pid)
-			case TagPort:
-				return buf.WriteString(c.Port())
-			case TagIP:
-				return buf.WriteString(c.IP())
-			case TagIPs:
-				return buf.WriteString(c.Get(fiber.HeaderXForwardedFor))
-			case TagHost:
-				return buf.WriteString(c.Hostname())
-			case TagPath:
-				return buf.WriteString(c.Path())
-			case TagURL:
-				return buf.WriteString(c.OriginalURL())
-			case TagUA:
-				return buf.WriteString(c.Get(fiber.HeaderUserAgent))
-			case TagLatency:
-				return buf.WriteString(fmt.Sprintf("%7v", stop.Sub(start).Round(time.Millisecond)))
-			case TagBody:
-				return buf.Write(c.Body())
-			case TagBytesReceived:
-				return appendInt(buf, len(c.Request().Body()))
-			case TagBytesSent:
-				return appendInt(buf, len(c.Response().Body()))
-			case TagRoute:
-				return buf.WriteString(c.Route().Path)
-			case TagStatus:
-				if cfg.enableColors {
-					return buf.WriteString(fmt.Sprintf("%s %3d %s", statusColor(c.Response().StatusCode(), colors), c.Response().StatusCode(), colors.Reset))
-				}
-				return appendInt(buf, c.Response().StatusCode())
-			case TagResBody:
-				return buf.Write(c.Response().Body())
-			case TagReqHeaders:
-				reqHeaders := make([]string, 0)
-				for k, v := range c.GetReqHeaders() {
-					reqHeaders = append(reqHeaders, k+"="+v)
-				}
-				return buf.Write([]byte(strings.Join(reqHeaders, "&")))
-			case TagQueryStringParams:
-				return buf.WriteString(c.Request().URI().QueryArgs().String())
-			case TagMethod:
-				if cfg.enableColors {
-					return buf.WriteString(fmt.Sprintf("%s %-7s %s", methodColor(c.Method(), colors), c.Method(), colors.Reset))
-				}
-				return buf.WriteString(c.Method())
-			case TagBlack:
-				return buf.WriteString(colors.Black)
-			case TagRed:
-				return buf.WriteString(colors.Red)
-			case TagGreen:
-				return buf.WriteString(colors.Green)
-			case TagYellow:
-				return buf.WriteString(colors.Yellow)
-			case TagBlue:
-				return buf.WriteString(colors.Blue)
-			case TagMagenta:
-				return buf.WriteString(colors.Magenta)
-			case TagCyan:
-				return buf.WriteString(colors.Cyan)
-			case TagWhite:
-				return buf.WriteString(colors.White)
-			case TagReset:
-				return buf.WriteString(colors.Reset)
-			case TagError:
-				if chainErr != nil {
-					return buf.WriteString(chainErr.Error())
-				}
-				return buf.WriteString("-")
-			default:
-				// Check if we have a value tag i.e.: "reqHeader:x-key"
+			if logFunc, ok := tagMap[tag]; ok {
+				return logFunc(buf, c, w, tag)
+			} else {
```

Comment:
can we use the feature also for these tags "reqHeader:x-key" and make the recognition of the tag additionally with the first part, if this was not found in the map?

then we can also define a function for "reqHeader" in the map and offer the users the possibility to create custom functions with dynamic elements

within the function we then give the real tag so that the information for the dynamic part can be extracted

but again please note the performance for this functionality

Reply:
ok